### PR TITLE
made realm name editable

### PR DESCRIPTION
**Pull request by edewit**
fixes: #941

### CODE REVIEW

File: src/context/realm-context/RealmContext.tsx [hunk 1]
```diff
@@ -45,21 +45,20 @@ export const RealmContextProvider: FunctionComponent = ({ children }) => {
...
   return (
     <RealmContext.Provider
       value={{
         realm,
         setRealm: set,
         realms,
         refresh: async () => {
+          //this is needed otherwise the realm find function will not return
+          //new or renamed realms because of the cached realms in the token (perhaps?)
+          await adminClient.keycloak?.updateToken(Number.MAX_VALUE);
```
Comment (jonkoops) [inline src/context/realm-context/RealmContext.tsx:61]: That's interesting :thinking:

File: src/realm-settings/RealmSettingsSection.tsx [hunk 2]
```diff
@@ -272,35 +297,33 @@ export const RealmSettingsSection = () => {
...
-                  >
-                    <KeysProvidersTab
-                      realmComponents={realmComponents}
-                      keyProviderComponentTypes={kpComponentTypes}
-                      refresh={refresh}
-                    />
-                  </Tab>
-                </Tabs>
-              )}
+                  <KeysListTab realmComponents={realmComponents!} />
```
Comment (jonkoops) [inline src/realm-settings/RealmSettingsSection.tsx:311]: If the aforementioned statement is changed there is no need to use a non-null assertion here.
```suggestion
                  <KeysListTab realmComponents={realmComponents} />
```

File: src/realm-settings/RealmSettingsSection.tsx [hunk 3]
```diff
@@ -272,35 +297,33 @@ export const RealmSettingsSection = () => {
...
+                </Tab>
+                <Tab
+                  id="providers"
+                  data-testid="rs-providers-tab"
+                  aria-label="rs-providers-tab"
+                  eventKey={1}
+                  title={<TabTitleText>{t("providers")}</TabTitleText>}
+                >
+                  <KeysProvidersTab
+                    realmComponents={realmComponents!}
```
Comment (jonkoops) [inline src/realm-settings/RealmSettingsSection.tsx:321]: If the aforementioned statement is changed there is no need to use a non-null assertion here.
```suggestion
                    realmComponents={realmComponents}
```

File: src/realm-settings/RealmSettingsSection.tsx [hunk 4]
```diff
@@ -317,24 +340,20 @@ export const RealmSettingsSection = () => {
...
-                  reset={() => resetForm(realm)}
-                  realm={realm}
-                />
-              )}
+              <LocalizationTab
+                key={key}
+                refresh={refresh}
+                save={save}
+                reset={() => resetForm(realm)}
+                realm={realm!}
```
Comment (jonkoops) [inline src/realm-settings/RealmSettingsSection.tsx:348]: If the aforementioned statement is changed there is no need to use a non-null assertion here.
```suggestion
                realm={realm}
```

File: src/realm-settings/RealmSettingsSection.tsx [hunk 5]
```diff
@@ -354,7 +373,10 @@ export const RealmSettingsSection = () => {
               aria-label="tokens-tab"
               title={<TabTitleText>{t("realm-settings:tokens")}</TabTitleText>}
             >
-              <RealmSettingsTokensTab reset={() => resetForm(realm)} />
+              <RealmSettingsTokensTab
+                realm={realm!}
```
Comment (jonkoops) [inline src/realm-settings/RealmSettingsSection.tsx:377]: If the aforementioned statement is changed there is no need to use a non-null assertion here.
```suggestion
                realm={realm}
```

File: src/realm-settings/TokensTab.tsx [hunk 6]
```diff
@@ -70,36 +69,15 @@ export const RealmSettingsTokensTab = ({
     javaKeystoreAlgOptions!
   );
 
-  const form = useForm<RealmRepresentation>();
-  const { control } = useFormContext();
+  const form = useFormContext<RealmRepresentation>();
+  const control = form.control;
```
Comment (jonkoops) [inline src/realm-settings/TokensTab.tsx:73]: You can use destructuring to shorten this statement. In fact we might actually want to create an ESLint rule for this (that's for another time).

```suggestion
  const { control } = form;
```

**Review by jonkoops — CHANGES_REQUESTED** [review]

File: src/context/realm-context/RealmContext.tsx (hunk 1 above)
Comment (edewit) [reply src/context/realm-context/RealmContext.tsx:61]: The token contains realm and access information if we don't refresh it after a realm has been created you get errors

**Review by jonkoops — APPROVED** [review]

**Comment by ssilvert** [comment]
13 Realm Settings tests are failing.  I'm re-running the tests but that probably won't fix the problem.

**Comment by jonkoops** [comment]
@stianst This is a known issue, looks like some of the review suggestions have broken the tests.

**Comment by edewit** [comment]
@ssilvert I thought I fixed it yesterday, but guess it's not enough I'll have another look

File: src/realm-settings/TokensTab.tsx [hunk 7]
```diff
@@ -70,36 +69,15 @@ export const RealmSettingsTokensTab = ({
...
   const offlineSessionMaxEnabled = useWatch({
     control,
     name: "offlineSessionMaxLifespanEnabled",
     defaultValue: realm?.offlineSessionMaxLifespanEnabled,
   });
 
-  const setupForm = (realm: RealmRepresentation) => {
-    const { ...formValues } = realm;
-    form.reset(formValues);
-    Object.entries(realm).map((entry) => {
```
Comment (jenny-s51) [inline src/realm-settings/TokensTab.tsx:85]: This was needed here in order to populate the last four fields of the form

**Review by jenny-s51 — COMMENTED** [review]
Looks and works great @edewit! However removing the `setupForm` from the TokensTab causes the last four fields of the tokens settings to not be populated with the data from the server

**Review by mfrances17 — APPROVED** [review]
LGTM - code looks fine, and testing worked as expected

File: src/realm-settings/TokensTab.tsx (hunk 7 above)
Comment (edewit) [reply src/realm-settings/TokensTab.tsx:85]: I moved it to the `RealmSettingsSection.tsx`

**Comment by jonkoops** [comment]
@edewit Looks like the tests are still failing? I've ran it a couple of times, but always seems to fail.

**Comment by edewit** [comment]
yes it seems that it doesn't save the `IdP account email verification` token is not saved, but it's also broken on master

**Review by jenny-s51 — APPROVED** [review]
Looks great!

**Review by jonkoops — APPROVED** [review]

**Pull request merged**